lib/getOptions.js

```diff
@@ -152,24 +149,41 @@ const getBaseOptions = cb => {
     options.parallel = true;
   }
 
-  // try to get the repo token as an environment variable
-  if (process.env.COVERALLS_REPO_TOKEN) {
-    options.repo_token = process.env.COVERALLS_REPO_TOKEN;
-  } else {
-    // try to get the repo token from a .coveralls.yml file
+  // load a .coveralls.yml file
+  const coveralls_yaml_conf = (() => {
     const yml = path.join(process.cwd(), '.coveralls.yml');
     try {
       if (fs.statSync(yml).isFile()) {
-        const coveralls_yaml_conf = yaml.safeLoad(fs.readFileSync(yml, 'utf8'));
-        options.repo_token = coveralls_yaml_conf.repo_token;
-        if (coveralls_yaml_conf.service_name) {
-          options.service_name = coveralls_yaml_conf.service_name;
-        }
+        return yaml.safeLoad(fs.readFileSync(yml, 'utf8'));
       }
     } catch (_) {
-      logger.warn('Repo token could not be determined.  Continuing without it. ' +
-                  'This is necessary for private repos only, so may not be an issue at all.');
+      logger.debug('No valid .coveralls.yml file found');
+    }
+  })();
+
+  // try to get repo token and service name from .coveralls.yml file
+  if (coveralls_yaml_conf) {
+    if (coveralls_yaml_conf.repo_token) {
+      options.repo_token = coveralls_yaml_conf.repo_token;
     }
+    if (coveralls_yaml_conf.service_name) {
+      options.service_name = coveralls_yaml_conf.service_name;
+    }
+  }
+
+  // try to get the repo token as an environment variable
+  if (process.env.COVERALLS_REPO_TOKEN) {
+    options.repo_token = process.env.COVERALLS_REPO_TOKEN;
+  }
+
+  if (!options.repo_token) {
```

Comment:
I believe we could just do a sanity check here for private repos instead of always showing this warning.
```suggestion
  if ('travis-pro' === options.service_name && !options.repo_token) {
```